pkg/bosh/manifest/data_gatherer.go

```diff
@@ -30,13 +31,17 @@ func NewDataGatherer(log *zap.SugaredLogger, manifest *Manifest) *DataGatherer {
 	}
 }
 
-// GatherData will collect different data
-// Collect job spec information
-// Collect job properties
-// Collect bosh links
-// Render the bpm yaml file data
-func (dg *DataGatherer) GatherData(baseDir string, cfOperatorNamespace string, instanceGroupName string) ([]byte, error) {
-	jobReleaseSpecs, jobProviderLinks, err := dg.CollectReleaseSpecsAndProviderLinks(baseDir, cfOperatorNamespace)
+// GenerateManifest will collect different data and return a deployment
+// manifest for an instance group.
+// See docs/rendering_templates.md#calculation-of-required-properties-for-an-instance-group-and-render-bpm
+//
+// Data gathered:
+// * job spec information
+// * job properties
+// * bosh links
+// * bpm yaml file data
+func (dg *DataGatherer) GenerateManifest(baseDir string, namespace string, instanceGroupName string) ([]byte, error) {
```

Comment:
Not sure if this is fully accurate. While using `manifest` could lead to misinterpretations. AFAIK, there is only one manifest, and that is the `desired manifest`(e.g. after applying ops files).  I liked to have there a reference to the term `datagather`, not sure if something like `GatherAndGenerateData` is more appropriate to the functionality of this cmd.

Reply:
My comment above should not stop this PR for being merged, just wanted to share some thoughts.